catalog/rest_test.go

```diff
@@ -0,0 +1,818 @@
+		r.Require().Equal(http.MethodGet, req.Method)
+
+		for k, v := range TestHeaders {
+			r.Equal(v, req.Header.Values(k))
+		}
+
+		w.WriteHeader(http.StatusNotFound)
+		json.NewEncoder(w).Encode(map[string]any{
+			"error": map[string]any{
+				"message": "Namespace does not exist: personal in warehouse 8bcb0838-50fc-472d-9ddb-8feb89ef5f1e",
+				"type":    "NoSuchNamespaceException",
+				"code":    404,
+			},
+		})
+	})
+
+	cat, err := catalog.NewRestCatalog("rest", r.srv.URL, catalog.WithOAuthToken(TestToken))
+	r.Require().NoError(err)
+
+	_, err = cat.ListNamespaces(context.Background(), catalog.ToRestIdentifier("accounting"))
+	r.ErrorIs(err, catalog.ErrNoSuchNamespace)
+	r.ErrorContains(err, "Namespace does not exist: personal in warehouse 8bcb0838-50fc-472d-9ddb-8feb89ef5f1e")
+}
+
+func (r *RestCatalogSuite) TestCreateNamespace200() {
+	r.mux.HandleFunc("/v1/namespaces", func(w http.ResponseWriter, req *http.Request) {
+		r.Require().Equal(http.MethodPost, req.Method)
+		r.Require().Equal("application/json", req.Header.Get("Content-Type"))
+
+		for k, v := range TestHeaders {
+			r.Equal(v, req.Header.Values(k))
+		}
+
+		defer req.Body.Close()
+		dec := json.NewDecoder(req.Body)
+		body := struct {
+			Namespace table.Identifier   `json:"namespace"`
+			Props     iceberg.Properties `json:"properties"`
+		}{}
+
+		r.Require().NoError(dec.Decode(&body))
+		r.Equal(table.Identifier{"leden"}, body.Namespace)
+		r.Empty(body.Props)
+
+		json.NewEncoder(w).Encode(map[string]any{
+			"namespace": []string{"leden"}, "properties": map[string]any{},
+		})
+	})
+
+	cat, err := catalog.NewRestCatalog("rest", r.srv.URL, catalog.WithOAuthToken(TestToken))
+	r.Require().NoError(err)
+
+	r.Require().NoError(cat.CreateNamespace(context.Background(), catalog.ToRestIdentifier("leden"), nil))
+}
+
+func (r *RestCatalogSuite) TestCreateNamespaceWithProps200() {
+	r.mux.HandleFunc("/v1/namespaces", func(w http.ResponseWriter, req *http.Request) {
+		r.Require().Equal(http.MethodPost, req.Method)
+		r.Require().Equal("application/json", req.Header.Get("Content-Type"))
+
+		for k, v := range TestHeaders {
+			r.Equal(v, req.Header.Values(k))
+		}
+
+		defer req.Body.Close()
+		dec := json.NewDecoder(req.Body)
+		body := struct {
+			Namespace table.Identifier   `json:"namespace"`
+			Props     iceberg.Properties `json:"properties"`
+		}{}
+
+		r.Require().NoError(dec.Decode(&body))
+		r.Equal(table.Identifier{"leden"}, body.Namespace)
+		r.Equal(iceberg.Properties{"foo": "bar", "super": "duper"}, body.Props)
+
+		json.NewEncoder(w).Encode(map[string]any{
+			"namespace": []string{"leden"}, "properties": body.Props,
+		})
+	})
+
+	cat, err := catalog.NewRestCatalog("rest", r.srv.URL, catalog.WithOAuthToken(TestToken))
+	r.Require().NoError(err)
+
+	r.Require().NoError(cat.CreateNamespace(context.Background(), catalog.ToRestIdentifier("leden"), iceberg.Properties{"foo": "bar", "super": "duper"}))
```

Comment:
can we also verify that the properties have actually be set on the namespace? Just verifying that there wasn't an error could be misleading and stuff could just silently pass

Reply:
This is all mocked in this test, there isn't any namespace to verify properties for. The `r.Equal` validations inside the handler verifies that the request was sent correctly and contained the properties as expected (see line 428)